src/cccid.rs

```diff
@@ -48,16 +48,27 @@ const CCC_TMPL: &[u8] = &[
     0x00, 0xfe, 0x00,
 ];
 
-/// Cardholder Capability Container (CCC) Identifier
+/// Cardholder Capability Container (CCC) Identifier card ID
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
-pub struct CCCID(pub [u8; YKPIV_CCCID_SIZE]);
+pub struct CccCardId(pub [u8; YKPIV_CCCID_SIZE]);
+
+/// Cardholder Capability Container (CCC) Identifier
+#[derive(Copy, Clone)]
+pub struct CCC(pub [u8; YKPIV_CCC_SIZE]);
+
+impl CCC {
+    /// Return CardId component of CHUID
+    pub fn cccid(&self) -> Result<CccCardId, Error> {
+        let mut cccid = [0u8; YKPIV_CCCID_SIZE];
+        cccid.copy_from_slice(&self.0[CCC_ID_OFFS..(CCC_ID_OFFS + YKPIV_CCCID_SIZE)]);
+        Ok(CccCardId(cccid))
+    }
 
-impl CCCID {
     /// Generate a random CCCID
-    pub fn generate() -> Result<Self, Error> {
+    pub fn generate() -> Result<CccCardId, Error> {
```

Comment:
The reason I think this is weird is because it's a method of `CCC`(ID).

If you want this to return a `CccCardId`, I think it should be made a method of that struct, still returning `Self`.